interfaces/cython/cantera/onedim.py

```diff
@@ -487,11 +484,30 @@ def from_pandas(self, df, restore_boundaries=True, settings=None):
         exported by `to_pandas`. The method uses `from_solution_array` and
         requires a working *pandas* installation. The package ``pandas`` can be
         installed using pip or conda.
-        """
+
+        Note: As `to_pandas` does not save information on boundary conditions, those
+        values cannot be restored automatically and have to be specified by the user.
+
+        .. deprecated:: 2.6
+           *pandas* input will no longer be supported after Cantera 2.6 as it is not
+           capable of recreating all information required to restore the object. As
+           alternatives, `pandas` dataframes can be read into a `SolutionArray` and
+           passed to the flame domain using `from_solution_array`. For automatic
```

Comment:
As a naive person, I don't see how this is different from direct pandas input. What does `SolutionArray` do that pandas can't do directly? Can we modify `to_pandas` to include the boundary information?

Reply:
I think that deprecating the method is preferable - it is essentially the same situation as for `write_csv` - we can write CSV files, but there is no direct way to import the data. 

All objects in the `onedim` category are composed of three domains: upstream/downstream plus the actual solution domain. Both `to_pandas` and `write_csv` only export the solution itself. From my perspective, if we decide to include boundary values here (which is doable) the interface will become a little more complex (we will have to save a row at the top/bottom that does not correspond to a location), with the actual implementation requiring handling of quite a few edge cases. The alternative is to just use what was specifically intended for saving/restoring full information on these objects (`save`/`restore` and `write_hdf`/`read_hdf`), which ultimately is a much safer route.

I'm not completely opposed to an implementation here, but I think what is done for pandas should also be done for CSV, and I am hesitant to roll out a new feature at the moment.

Reply:
If it's a bug/broken, then it should be fixed or outright removed, IMO. If it's just not going to work at all, why bother warning? Just throw a clear error with the solution and be done with it.

~That said, I don't see how passing the output of `to_pandas` through `from_solution_array` fixes the problem of not having the boundary conditions.~ On rereading, I guess you're saying that the user should set up the `FreeFlame` and then pass the `SolutionArray` created from pandas to specifically set the state of the `flame` domain within the `FreeFlame`? I also see that this method is doing that, so why do we need to deprecate/remove this method? I think it would make more sense to ensure that the boundary conditions are set before loading, but still allow this behavior of loading from pandas DataFrames as long as that is satisfied.

Can you mention some of the edge cases you thought of in implementing `from_pandas`/`read_csv` with boundaries?

> All objects in the `onedim` category are composed of three domains: upstream/downstream plus the actual solution domain.

In principle, they can be composed of an arbitrary number of domains, I think; as a practical matter, the most convenient ones that we've implemented are composed of three domains 😄 

> I'm not completely opposed to an implementation here, but I think what is done for pandas should also be done for CSV, and I am hesitant to roll out a new feature at the moment.

I agree with this, but let's not shoot ourselves in the foot by removing things only to add them back later.

Reply:
> Can you mention some of the edge cases you thought of in implementing `from_pandas`/`read_csv` with boundaries?

Some of this was implemented prior to #840 (a working implementation can be seen in the deleted code segments of that PR, mostly in `FlameBase.to_solution_array` and `FlameBase.from_solution_array`) ... in order for this approach to work, compatible output needs to include rows that are not grid points (i.e. CSV and pandas output need to be prepped). I did not retain this as it won't work for `Surface` interactions (see `catalytic_combustion.py` example).